.circleci/config.yml

```diff
@@ -168,7 +168,7 @@ jobs:
   wp-desktop-mac:
     macos:
       xcode: '14.2.0'
-    resource_class: macos.x86.medium.gen2
+    resource_class: macos.m1.medium.gen1
```

Comment:
@nsakaimbo What do you think? Is there a cost penalty incurred to use the M1? 

Performance wise, we see a nice speed-up with the M1 builders. 
Compare the below:

![image](https://github.com/Automattic/wp-calypso/assets/6549265/d69f9b8d-facc-466e-a569-57366c59653d)

The left shows the M1 builder and the right shows the Intel builder.
Even though the M1 builder is building both the x64 and arm64 binaries, the overall execution time for the build is the same as the Intel builder, which is only building one binary.

Should we go back to building only one binary by default, this would mean the overall task execution time being cut down by as much as a minute.

Reply:
That's great to see that the M1 image is a lot more performant. 👏🏽  I'm not sure about any cost penalty, though - to my knowledge, we are only able to use images that are accessible with our existing Circle plan, so I think this should be OK?

> Should we go back to building only one binary by default, this would mean the overall task execution time being cut down by as much as a minute.

Yeah I think we should limit the binaries generated during non-release runs to just one (see my other comment on this point).